parl/remote/utils.py

```diff
@@ -52,49 +54,115 @@ def data_process():
     return to_write_lines
 
 
-def load_remote_class(file_name, class_name, end_of_file, in_sys_path):
+def is_class_defined_in_file(cls):
+    """Check whether the class is defined in a python file
+
+    Args:
+        cls: class
     """
-  load a class given its file_name and class_name.
+    if hasattr(cls, '__module__'):
+        cls_module = sys.modules.get(cls.__module__)
+        if getattr(cls_module, '__file__', None):
+            return True
+    return False
 
-  Args:
-    file_name: specify the file to load the class
-    class_name: specify the class to be loaded
-    end_of_file: line ID to indicate the last line that defines the class.
-    in_sys_path: whether the path of the remote class is in the environment path (sys.path).
 
```

Comment:
There are too many functions in `utils.py`. 
I suggest opening a new file named `remote_class_distribution` for these implementation.

Reply:
remote_class_serialization

Reply:
fixedd